pkg/util/httprequest/httprequest.go

```diff
@@ -68,43 +68,52 @@ func SchemeHost(req *http.Request) (string /*scheme*/, string /*host*/) {
 
 	forwardedProto := forwarded("Proto")
 	forwardedHost := forwarded("Host")
-	// If both X-Forwarded-Host and X-Forwarded-Port are sent, use the explicit port info
-	if forwardedPort := forwarded("Port"); len(forwardedHost) > 0 && len(forwardedPort) > 0 {
-		if h, _, err := net.SplitHostPort(forwardedHost); err == nil {
-			forwardedHost = net.JoinHostPort(h, forwardedPort)
-		} else {
-			forwardedHost = net.JoinHostPort(forwardedHost, forwardedPort)
-		}
+	forwardedPort := forwarded("Port")
+
+	scheme := ""
```

Comment:
Me too, but you want [https://github.com/openshift/origin/pull/14319#discussion_r119357611](determine the scheme by port), and I want to not add the port if it's a default port for the scheme.